### PR TITLE
Fix comments, doctrings and type annotations

### LINKED ISSUE #419: Fix comments which lie about test file contents etc.
There are various places where comments have diverged from reality, and therefore mislead the reader. We notice these while working on all sorts of things, but never get around to fixing them.

This issue should serve as a place where we can note any such comments we find. Once a signifcant number of such comments has been noted here, We can fix a whole bunch of them in one go.

Please note any misleading comments you find (and do not have the time to fix immediately) by commenting in this issue.

### PR DESCRIPTION
**Pull request by carmenromo**
There are various places where comments have diverged from reality, and therefore mislead the reader. This PR fix some comments, doctrings and type annotations and also serves to remind other users to contribute information about what else needs to be fixed. 

closes #419

### CODE REVIEW

**Comment by carmenromo** [comment]
@jacg please could you take a look?

**Comment by jacg** [comment]
In some places you use `List` to annotate parameter types. I suspect that this is not true. I suspect that you are being too restrictive and not giving a sufficiently liberal Duck Type. Are you sure it's not sequence?

Please add a `closes #NNN` to the description, replacing `NNN` with the number of the relevant issue. Like this, the issue will be closed automatically when this PR is merged.

You should ask someone who is intimately familiar with the code to review your changes for each bit. `git blame` and `M-x magit-blame` could help you identify the relevant people.

**Comment by carmenromo** [comment]
@jacg I had the doubt of List or Sequence but I found in the python documentation this:

class typing.List(list, MutableSequence[T])
Generic version of list. Useful for annotating return types. To annotate arguments it is preferred to use abstract collection types such as Mapping, Sequence, or AbstractSet.

**Comment by carmenromo** [comment]
@gonzaponte @paolafer please could you take a look?

**Comment by jacg** [comment]
@carmenromo Why don't you have a go at rebasing this, and resolving the merge conflicts. As this isn't crucial or urgent, it's a nice toy problem to work with. And activity in this PR might even encourage some of us to look at the work you did and maybe even ... approve the PR!

**Comment by jjgomezcadenas** [comment]
Yes, please. Rebase and solve conflicts, then either @paolafer or @gonzaponte, who has been in limbo long enough, can take a look to this.

**Comment by jacg** [comment]
You seem to have merged the `master` branch into your feature branch. That's not how we do things 'round here. There are two options

1. To include, into your feature branch, new code that has been added to the central `master` branch, *rebase* your feature branch on top of it.
1. To include the work that you did on your feature branch into the central `master` branch, you should merge your feature branch into `master`.

The second of these only happens after the PR containing your feature branch has been approved. The first one, you can do whenever you feel like it, and you *must* do it before the second step.

So, you need to undo the merge you just did. You do this by resetting your feature branch to the commit that precedes the merge commit you just created (or the feature branch side, rather than the `master` side).

**Comment by jacg** [comment]
One more thing: after you reset your feature branch, an ordinary push to your fork will not work. You will need to push with the `--force-with-lease` option, or just `f` (lowercase, not uppercase) in Magit.

**Comment by jjgomezcadenas** [comment]
@carmenromo please rebase. 
@bpalmeiro please review

File: invisible_cities/io/mcinfo_io.py [hunk 1]
```diff
@@ -219,6 +219,10 @@ def read_mcinfo_evt (mctables: (tb.Table, tb.Table, tb.Table, tb.Table),
             while ipart <= ipart_end:
                 particle_rows.append(h5particles[ipart])
                 ipart += 1
+            
+            # It is possible for the 'generators' dataset to have a different length compared to the 'extents' dataset. In particular, it may occur that the 'generators' dataset is empty. In this case, do not add any rows to 'generators'.
+            if len(h5generators) == len(h5extents):
+                generator_rows.append(h5generators[last_row])
 
             # It is possible for the 'generators' dataset to have a different length compared to the 'extents' dataset. In particular, it may occur that the 'generators' dataset is empty. In this case, do not add any rows to 'generators'.
```
Comment (bpalmeiro) [inline invisible_cities/io/mcinfo_io.py]: This piece of code is duplicated. This is also the reason of the test failure.

**Review by bpalmeiro — CHANGES_REQUESTED** [review]
Great job! just minor issue pending

File: invisible_cities/io/mcinfo_io.py (hunk 1 above)
Comment (carmenromo) [reply invisible_cities/io/mcinfo_io.py]: Ups you are right!

**Review by bpalmeiro — APPROVED** [review]

**Comment by bpalmeiro** [comment]
Perfect! Please, rebase the branch and it is ready to be merged.

**Comment by carmenromo** [comment]
@jacg how can I move this PR to the _merge_ column?

**Comment by jacg** [comment]
> how can I move this PR to the merge column?

Drag'n'drop.

Of course, it doesn't need to be in the *merge* column, for you to be able to do the merge. So if the next thing you are going to do is merge, then you'll be forgiven for not moving it to the *merge* columnn: just merge, and it will automatically be moved to the *done* column.

**Pull request merged**